### PR TITLE
[core][autoscaler] Stream autoscaler events through log monitor in v2

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
With autoscaler v2, we will no longer have `monitor.log` if autoscaler is not deployed together with ray.  In v1, we will stream the "event_summary" lines in the `monitor.log` to driver. 

In order to stay parity with legacy behaviour (having autoscaler events printed out in the driver), we will stream the autoscaler events (which are the same content that we were streaming to driver in legacy behaviour). 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/event/event_logger.py [hunk 1]
```diff
@@ -138,3 +138,18 @@ def get_event_logger(source: Event.SourceType, sink_dir: str):
...
+
+    Args:
+        event_str: The string to parse.
+
+    Returns:
+        The parsed event if parsable, else None
+    """
+    try:
+        return Parse(event_str, Event())
+    except Exception:
```
Comment (iycheng) [inline python/ray/_private/event/event_logger.py:154]: This seems strange. In which case this should happen? Should we log error?

File: python/ray/_private/event/event_logger.py [hunk 2]
```diff
@@ -138,3 +138,18 @@ def get_event_logger(source: Event.SourceType, sink_dir: str):
             _event_logger[source_name] = EventLoggerAdapter(source, logger)
 
         return _event_logger[source_name]
+
+
+def parse_event(event_str: str) -> Optional[Event]:
+    """Parse an event from a string.
+
+    Args:
+        event_str: The string to parse.
```
Comment (rkooo567) [inline python/ray/_private/event/event_logger.py:147]: can you comment the string format in the docstring?

File: python/ray/_private/log_monitor.py [hunk 3]
```diff
@@ -194,27 +211,31 @@ def _close_all_files(self):
...
+        ) + glob.glob(f"{self.logs_dir}/java-worker*.log")
         # segfaults and other serious errors are logged here
-        raylet_err_paths = glob.glob(f"{self.logs_dir}/raylet*.err")
+        monitor_log_paths += glob.glob(f"{self.logs_dir}/raylet*.err")
         # monitor logs are needed to report autoscaler events
-        monitor_log_paths = glob.glob(f"{self.logs_dir}/monitor.log")
+        # TODO(rickyx): remove this after migration.
+        if not self.is_autoscaler_v2:
+            # We publish monitor logs in autoscaler v1
+            monitor_log_paths += glob.glob(f"{self.logs_dir}/monitor.log")
```
Comment (rkooo567) [inline python/ray/_private/log_monitor.py:225]: Q: Should we just print event_AUTOSCALER.log for both v1 & v2? Isn't v1 also writing all events to this file?

File: python/ray/_private/log_monitor.py [hunk 4]
```diff
@@ -329,6 +350,8 @@ def flush():
                     "task_name": file_info.task_name,
                 }
                 try:
+                    if "AUTOSCALER" in file_info.filename:
```
Comment (rkooo567) [inline python/ray/_private/log_monitor.py:353]: remove it?

File: python/ray/_private/log_monitor.py [hunk 5]
```diff
@@ -343,6 +366,7 @@ def flush():
             for _ in range(max_num_lines_to_read):
                 try:
                     next_line = file_info.file_handle.readline()
+                    logger.info(f"file_info: {repr(file_info)}\nnext_line: {next_line}")
```
Comment (rkooo567) [inline python/ray/_private/log_monitor.py:369]: remove it?

File: python/ray/_private/worker.py [hunk 6]
```diff
@@ -1783,26 +1784,52 @@ def print_to_stdstream(data):
...
-                    "messages, set RAY_SCHEDULER_EVENTS=0."
-                )
-                autoscaler_log_fyi_printed = True
-            # The event text immediately follows the ":event_summary:"
-            # magic token.
-            yield line.split(ray_constants.LOG_PREFIX_EVENT_SUMMARY)[1]
+    AUTOSCALER_LOG_FYI = (
+        "Tip: use `ray status` to view detailed "
+        "cluster status. To disable these "
+        "messages, set RAY_SCHEDULER_EVENTS=0."
```
Comment (rkooo567) [inline python/ray/_private/worker.py:1800]: does v2 still work with this env var?

Comment (rkooo567) [reply python/ray/_private/worker.py:1800]: can we add a test?

**Review by rkooo567 — CHANGES_REQUESTED** [review]
Do you think it is feasible to just use event logs for even v1? 

Also is it already tested somewhere? Or do you plan to add tests?

**Comment by rickyyx** [comment]
> Do you think it is feasible to just use event logs for even v1?
> 
> Also is it already tested somewhere? Or do you plan to add tests?

Yeah, use this for v1 is also viable. 

I am gonna add tests.

**Comment by rickyyx** [comment]
Updates:
- Added tests 
- Feature flag it for v2 still (not going to change v1's code since we are likely gonna migrate it in 2.7, and supporting v1+v2 together would also require much more testing with v1)

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**